google-cloud-pubsub/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,182 @@
+2. **Removal of handwritten resource management APIs**: These existing APIs will be replaced with the 
+gRPC-based clients in `google-cloud-pubsub-v1`. This deprecation includes CRUDL operations for topics, subscriptions,
+and schemas as well as IAM.
+    1. Handwritten classes such as Snapshot, Schema, and ReceivedMessage are removed in favor of their auto-generated counterparts.
+3. **Restructed Publishing and Subscribing**:
+    1. To better reflect their purpose, the Publisher and Subscriber objects now exclusively handle the publishing and receiving
+    of messages. These were formerly done via Topic and Subscription, respectively.
+    2. The existing Subscriber class was renamed MessageListener.
+4. **gRPC Client Exposure**: As a result of removing handwritten admin operations, `TopicAdmin::Client`, `SubscriptionAdmin::Client`
+and `SchemaService::Client` are now accessible directly from `Google::Cloud::PubSub::Project`.
+
+## Legacy Support
+
+The existing `google-cloud-pubsub` (henceforth known as the v2 library) will be supported for one year from when the `google-cloud-pubsub` v3
+is released, giving developers proper time to migrate. Features will be prioritized in the new library, while still providing security and
+bug fixes in v2.
+
+## Data Plane Operations: Publishing and Receiving Messages
+
+In the new version of the library, publishing and receiving messages are handled separately from managing resources.
+
+### Publishing messages
+
+Previously, publishing was performed by retrieving a Topic object and calling publish on it. This is now done by performing the
+same action on a Publisher.
+
+```ruby
+# Before
+pubsub = Google::Cloud::Pubsub.new project_id: project_id
+
+topic = pubsub.topic "my-topic"
+
+topic.publish "This is a test message."
+```
+
+```ruby
+# After
+pubsub = Google::Cloud::Pubsub.new project_id: project_id
+
+publisher = pubsub.publisher "my-topic"
+
+publisher.publish "This is a test message."
+```
+
+
+### Receiving Messages
+
+Similarly, a Subscription object no longer manages receiving messages; that action is now performed by a Subscriber.
+
+```ruby
+# Before
+pubsub = Google::Cloud::PubSub.new project_id: project_id
+
+subscription = pubsub.subscription "my-topic-sub"
+
+subscriber = subscription.listen do |received_message|
+  puts "Message: #{received_message.message.data}"
+  received_message.acknowledge!
+end
+
+subscriber.start
+
+```
+
+```ruby
+# After
+pubsub = Google::Cloud::PubSub.new project_id: project_id
+
+subscriber = pubsub.subscriber "my-topic-sub"
+
+listener = subscriber.listen do |received_message|
+  puts "Message: #{received_message.message.data}"
+  received_message.acknowledge!
+end
+
+listener.start
+```
+
+## Admin Plane Operations: Managing Resources
+
+With the removal of all hand-written resource management methods, users will now rely on using the auto-generated gRPC clients for all administrative tasks.
```

Comment:
We should remain consistent with terms. We use admin mostly everywhere else, administrative is a deviation.

```suggestion
With the removal of all hand-written resource management methods, users will now rely on using the auto-generated gRPC clients for all admin plane calls.
```